### PR TITLE
Fix for issue #591.  TreeEntry.path() throws when TreeEntry came from Tree.entries()

### PR DESCRIPTION
**Pull request by tomruggs**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -74,7 +74,8 @@
     "nw-gyp": "^0.12.4",
     "pangyp": "^2.1.0",
     "request": "^2.55.0",
-    "tar": "^2.1.0"
+    "tar": "^2.1.0",
+    "bluebird": "^2.9.26"
```
Comment (tbranyen) [inline package.json]: This dependency isn't needed.  We use `nodegit-promise`.

File: test/tests/tree_entry.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 var assert = require("assert");
+var promise = require("bluebird");
```
Comment (tbranyen) [inline test/tests/tree_entry.js]: `var Promise = require('nodegit-promise')`

File: package.json (hunk 1 above)
Comment (tomruggs) [inline package.json]: Thanks, not sure how I missed that.  Removed.

**Comment by tbranyen** [comment]
Looks good, thanks!

**Pull request merged**